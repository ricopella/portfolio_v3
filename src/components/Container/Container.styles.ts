import { WIDTHS } from '../../styles/variables'
import styled from '@emotion/styled'

const StyledContainer = styled.div`
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: auto;
  max-width: ${WIDTHS.lg};
`

export default StyledContainer
