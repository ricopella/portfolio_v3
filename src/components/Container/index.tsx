import React, { FC } from 'react'
import StyledContainer from './Container.styles'
import { ContainerProps } from '../../types'

const Container: FC<ContainerProps> = ({ children, className, style }) => (
  <StyledContainer className={className} style={style}>
    {children}
  </StyledContainer>
)

export default Container
