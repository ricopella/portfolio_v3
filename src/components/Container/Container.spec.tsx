import renderer from 'react-test-renderer'
import React from 'react'
import Container from './index'
import { render } from '@testing-library/react'

describe('<Container />', () => {
  it('should match Container snapshot', () => {
    const internalTree = renderer
      .create(
        <Container>
          <div>Wu-Tang Clan</div>
        </Container>
      )
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { getByText } = render(
      <Container>
        <div>Wu-Tang Clan</div>
      </Container>
    )

    expect(getByText(/Wu-Tang Clan/i)).toBeTruthy()
  })
})
