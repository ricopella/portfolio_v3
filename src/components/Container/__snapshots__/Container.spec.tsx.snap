// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match Container snapshot 1`] = `
<div
  className="css-wvjbft"
>
  <div>
    Wu-Tang Clan
  </div>
</div>
`;
